catalog/confluence/catalog-confluence-source/src/main/java/org/codice/ddf/confluence/source/ConfluenceSource.java

```diff
@@ -351,15 +365,74 @@ public void setBodyExpansion(String bodyExpansion) {
 
   private Result getUpdatedResult(Metacard metacard) {
     metacard.setSourceId(this.getId());
-    for (Map.Entry<String, Set<String>> entry : attributeOverrides.entrySet()) {
-      Set<String> value = entry.getValue();
-      if (value.size() == 1) {
-        metacard.setAttribute(new AttributeImpl(entry.getKey(), value.iterator().next()));
-      } else {
-        metacard.setAttribute(new AttributeImpl(entry.getKey(), new ArrayList<String>(value)));
+    overrideAttributes(metacard, attributeOverrides);
+
+    return new ResultImpl(metacard);
+  }
+
+  private void overrideAttributes(Metacard metacard, Map<String, Set<String>> attributeOverrides) {
+    if (MapUtils.isEmpty(attributeOverrides)) {
+      return;
+    }
+
+    attributeOverrides
+        .keySet()
+        .stream()
+        .map(attributeRegistry::lookup)
+        .filter(Optional::isPresent)
+        .map(Optional::get)
+        .map(ad -> overrideAttributeValue(ad, attributeOverrides.get(ad.getName())))
+        .filter(Objects::nonNull)
+        .forEach(metacard::setAttribute);
+  }
+
+  private AttributeImpl overrideAttributeValue(
+      AttributeDescriptor attributeDescriptor, Set<String> overrideValue) {
+    List<Serializable> newValue = new ArrayList<>();
+    for (String override : overrideValue) {
+      try {
+        switch (attributeDescriptor.getType().getAttributeFormat()) {
+          case INTEGER:
+            newValue.add(Integer.parseInt(override));
+            break;
+          case FLOAT:
+            newValue.add(Float.parseFloat(override));
+            break;
+          case DOUBLE:
+            newValue.add(Double.parseDouble(override));
+            break;
+          case SHORT:
+            newValue.add(Short.parseShort(override));
+            break;
+          case LONG:
+            newValue.add(Long.parseLong(override));
+            break;
+          case DATE:
+            Calendar calendar = DatatypeConverter.parseDateTime(override);
+            newValue.add(calendar.getTime());
+            break;
+          case BOOLEAN:
+            newValue.add(Boolean.parseBoolean(override));
+            break;
+          case BINARY:
+            newValue.add(override.getBytes(Charset.forName("UTF-8")));
+            break;
+          case OBJECT:
+          case STRING:
+          case GEOMETRY:
+          case XML:
+            newValue.add(override);
+            break;
+        }
+      } catch (IllegalArgumentException e) {
```

Comment:
❓ Do we need to account for any other type of exception handling for defensive purposes and test accordingly? For example if some reason the attributeFormat is Integer but the override value causes a NumberFormatException? Although worst case here is the single attribute not overridden?

Reply:
NumberFormatException is an instance of IllegalArgumentException so that case would be covered here. This is the same logic used in two other places in the code base for attribute overrides so I hope it is ok as is.